### PR TITLE
Strip all Type System Directives during composition

### PR DESCRIPTION
**Pull request by josemarluedke**
Fix #3565.

### CODE REVIEW

**Comment by trevor-scheer** [comment]
Thanks for taking the time to put this PR together, @josemarluedke! This is pretty close to what we want, I've created a [few commits](https://github.com/apollographql/apollo-server/tree/trevor/update-josemarluedke-fix-3565) and pushed them up to show how I'd expand the scope of this a bit. Feel free to take inspiration from or cherry-pick them altogether.

In short, `FIELD_DEFINITION` directives aren't the only type of custom directives we want to remove from the SDL. We're interested in _all_ custom directives.

Let me know your thoughts and when you're ready for another review. I'm also happy to land these changes myself and maintain your authorship - let me know what works best for you!

A changelog entry would also be appreciated:
https://github.com/apollographql/apollo-server/blob/master/packages/apollo-federation/CHANGELOG.md

Thanks again 😄

**Comment by josemarluedke** [comment]
@trevor-scheer Thanks for creating these commits. I have cherry-picked them here. I also added an entry to the federation changelog. 

Let me know if anything is missing, but I believe this is good to go.

File: packages/apollo-federation/src/composition/__tests__/composeAndValidate.test.ts [hunk 1]
```diff
@@ -551,3 +551,76 @@ describe('composition of value types', () => {
...
+   * merely for use by the type-system ("type-system"), see the GraphQL spec:
+   * https://graphql.github.io/graphql-spec/June2018/#sec-Type-System.Directives
+   */
+  it('preserves executable and purges type-system directives', () => {
+    const serviceA = {
+      typeDefs: gql`
+        "directives at FIELDs are executable"
+        directive @audit(risk: Int!) on FIELD
+
+        "directives at FIELD_DEFINITIONs are for the type-system"
```
Comment (zionts) [inline packages/apollo-federation/src/composition/__tests__/composeAndValidate.test.ts]: It seems that `@deprecated` defies this semantic: https://www.apollographql.com/docs/graphql-tools/schema-directives/

**Comment by glasser** [comment]
We think this PR may accidentally be stripping `@deprecated`.

**Comment by abernix** [comment]
Correct assessment above by @glasser in https://github.com/apollographql/apollo-server/pull/3736#issuecomment-586014813 and fixed by https://github.com/apollographql/apollo-server/pull/3792

**Comment by XBeg9** [comment]
@josemarluedke any ideas why we need to strip directives from service schema (introspection) itself? it's hard for service-level-engineer to work with their own schema, moreover, how to track changes to the schema? Probably we can strip all directives from SDL, but not from schema itself. Any concerns?
Thanks

**Pull request merged**